### PR TITLE
[RW-8041][risk=low]Added the ability to use the custom request access url from the regis…

### PR DESCRIPTION
**Pull request by evrii**
Description: Added the ability to use the custom request access url from the registration page when a registrant attempts to register for an institute that requires only specific email addresses.

https://user-images.githubusercontent.com/24514630/159502948-0d6cdff7-b330-4ba3-8e45-9d01dbadd96f.mov

https://user-images.githubusercontent.com/24514630/159574644-f0a9f05b-eb75-428e-af7e-937c7e9950be.mov

---
**PR checklist**

- [x] This PR meets the Acceptance Criteria in the JIRA story
- [x] The JIRA story has been moved to Dev Review
- [ ] This PR includes appropriate unit tests
- [x] I have run and tested this change locally
- [ ] I have run the E2E tests on ths change against my local UI and/or API server with `yarn test-local` or [yarn test-local-devup](https://github.com/all-of-us/workbench/blob/main/e2e/README.md#examples)
- [x] If this includes a UI change, I have taken screen recordings or screenshots of the new behavior and notified the PO and UX designer
- [x] If this includes an API change, I have updated the appropriate Swagger definitions and notified API consumers
- [ ] If this includes a new feature flag, I have created and linked new JIRA tickets to (a) turn on the feature flag and (b) remove it later

### CODE REVIEW

File: api/src/main/resources/workbench-api.yaml [hunk 1]
```diff
@@ -6913,6 +6913,9 @@ definitions:
         "$ref": "#/definitions/InstitutionMembershipRequirement"
         description: 'Type of Registered tier requirement the institution has signed, email domains
          or restricted to a defined set of specific researchers'
+      requestAccessUrl:
+        type: string
+        description: Request access url for users to use during registration.
```
Comment (jmthibault79) [inline api/src/main/resources/workbench-api.yaml:6918]: please note that this is optional

File: ui/src/app/utils/institutions.tsx [hunk 2]
```diff
@@ -42,16 +43,22 @@ export async function checkInstitutionalEmail(
   }
 }
 
-const EmailAddressMismatchErrorMessage = () => {
+const EmailAddressMismatchErrorMessage = ({ requestAccessUrl }) => {
```
[the remark this answers is not in the document]
Comment (jmthibault79) [reply ui/src/app/utils/institutions.tsx:46]: this is a nice extension of the existing version.

**Review by jmthibault79 — COMMENTED** [review]
Looks good so far.  There is one piece that remains however: we need to also use this link for the "Request Access" to Controlled Tier button on the Data Access Requirements page:
Screen Shot 2022-03-22 at 1 58 25 PM

File: api/src/main/resources/workbench-api.yaml (hunk 1 above)
Comment (evrii) [reply api/src/main/resources/workbench-api.yaml]: I updated the description. Thanks for pointing this out.

**Comment by evrii** [comment]
> Looks good so far. There is one piece that remains however: we need to also use this link for the "Request Access" to Controlled Tier button on the Data Access Requirements page: Screen Shot 2022-03-22 at 1 58 25 PM

Thanks for pointing this out. I added the functionality to work on this page as well. I updated the description of the PR to include a recording of how this screen works as well.

File: api/src/main/resources/workbench-api.yaml [hunk 3]
```diff
@@ -6878,7 +6878,7 @@ definitions:
           if its enumerated type is OTHER'
       requestAccessUrl:
         type: string
-        description: Request access url for users to use during registration.
+        description: Optional url for institution's handling their own access requests
```
Comment (jmthibault79) [inline api/src/main/resources/workbench-api.yaml:6881]: sometimes apostrophes here can cause problems.  did you notice anything running locally?

File: ui/src/app/components/support.tsx [hunk 4]
```diff
@@ -10,8 +12,21 @@ export const SupportMailto = ({ label = SUPPORT_EMAIL, style = {} }) => (
   </StyledExternalLink>
 );
 
-export const SupportButton = ({ label = SUPPORT_EMAIL, style = {} }) => (
-  <Button style={style} onClick={() => window.open(`mailto:${SUPPORT_EMAIL}`)}>
+const handleClickSupportButton = (url) => () => {
+  if (url) {
```
Comment (jmthibault79) [inline ui/src/app/components/support.tsx:16]: this logic is now repeated - would it make sense to reuse some code here?

File: ui/src/app/pages/homepage/data-access-requirements.tsx [hunk 5]
```diff
@@ -952,7 +955,10 @@ const ControlledTierCard = (props: {
               is required.
             </div>
             <div style={styles.requestAccess}>
-              <SupportButton label='Request Access' />
+              <SupportButton
```
Comment (jmthibault79) [inline ui/src/app/pages/homepage/data-access-requirements.tsx:958]: I like how concise this is.  But I'm not sure whether this use case should live inside SupportButton - it seems to me like it's pretty specific to D-A-R.  What do you think?

File: api/src/main/resources/workbench-api.yaml (hunk 3 above)
Comment (jmthibault79) [reply api/src/main/resources/workbench-api.yaml:6881]: (I did not)

**Review by jmthibault79 — APPROVED** [review]
Tested locally.  Works great.

Very nice recordings.

Your DB workflow looks nicer than mine - I'd love to learn more about that sometime.

File: api/src/main/resources/workbench-api.yaml (hunk 3 above)
Comment (evrii) [reply api/src/main/resources/workbench-api.yaml:6881]: Good point. I do not recall seeing any issues. Agreed though, apostrophes can be tricky.

File: ui/src/app/components/support.tsx (hunk 4 above)
Comment (evrii) [reply ui/src/app/components/support.tsx:16]: Yeah, I wonder if this functionality should live in the urls utils file.

File: ui/src/app/pages/homepage/data-access-requirements.tsx (hunk 5 above)
Comment (evrii) [reply ui/src/app/pages/homepage/data-access-requirements.tsx:958]: Good point. I did notice that SupportButton is only used in D-A-R. I am good with moving it.

File: ui/src/app/utils/urls.ts [hunk 6]
```diff
@@ -9,3 +9,23 @@ export function canonicalizeUrl(url: string): string {
...
+function isValidUrl(potentialUrl: string): boolean {
+  try {
+    new URL(potentialUrl);
+    return true;
+  } catch (e) {
+    return false;
+  }
+}
+
+export function getCustomOrDefaultUrl(customUrl: string, defaultUrl: string) {
```
Comment (evrii) [inline ui/src/app/utils/urls.ts:22]: I am unsure of the name of this function. I am open to suggestions if anyone can think of anything better.

**Review by NehaBroad — APPROVED** [review]
Test for ByPass/Expiring Soon and Expired

**Review by NehaBroad — APPROVED** [review]
Tested 👍

**Pull request merged**